src/lib/containers/widgets/facet-nav/FacetNavPanel.tsx

```diff
@@ -368,27 +368,27 @@ const FacetNavPanel: React.FunctionComponent<FacetNavPanelProps> = (
     [data, facetToPlot.columnName],
   )
 
-  useEffect(() => {
-    let isMounted = true
-    if (!facetToPlot) {
-      return
-    } else {
+  const { data: plotData } = useQuery(
+    [
+      'extractPlotDataArray',
+      facetToPlot,
+      getColumnType(),
+      index,
+      plotType,
+      accessToken,
+    ],
+    () =>
       extractPlotDataArray(
         facetToPlot,
         getColumnType(),
         index,
         plotType,
         accessToken,
-      ).then(plotData => {
-        if (isMounted) {
-          setPlotData(plotData)
-        }
-      })
-    }
-    return () => {
-      isMounted = false
-    }
-  }, [facetToPlot, data, index, plotType, accessToken, getColumnType])
+      ),
+    {
+      enabled: !!facetToPlot,
+    },
+  )
```

Comment:
One small logic change; the `useEffect` + `isMounted` pattern can cause cause delayed updates which triggers a warning in tests. react-query has already solved this general problem, so let's just wrap our async call in `useQuery`.